m4/check_pkg_cuda.m4

```diff
@@ -15,16 +15,31 @@ AC_DEFUN([CHECK_PKG_CUDA], [
   AC_ARG_WITH([cuda],
      [AS_HELP_STRING([--with-cuda=PATH], [Path to non-standard CUDA installation])])
 
+  AC_MSG_CHECKING([if dynamically linking cuda is requested])
+  AC_ARG_ENABLE([cudart-dynamic],
+    [AS_HELP_STRING([--enable-cudart-dynamic],
+                    [link cudart dynamically (default=link statically)])],,
+    [enable_cudart_dynamic=no])
+  enable_cudart_dynamic=`echo $enable_cudart_dynamic`
+  case $enable_cudart_dynamic in
```

Comment:
This seems a bit overcomplicated, but it works, so *shrug*

Reply:
There's a lot of different ways to supply a single flag:

* --enable-foo
* --disable-foo
* --enable-foo=yes
* --enable-foo=no
* --disable-foo=no
* --disable-foo=yes

I've noticed on occasion, for at least some of our options, that `./configure` will happy accept a subset of these options, but then the build doesn't actually do the right thing. Particularly with the bare flags. Point blank: I do not trust myself to use `AC_ARG_ENABLE` in a way that is both concise and correct.